### PR TITLE
Ai database storage

### PR DESCRIPTION
**Pull request by Lewis-Kelley**
The AI game now successfully stores all actions to the database. Note that this does __not__ seek to change the schema, though many of the refactors should help with that.

There were some changes to the formatting of the output, but I don't think they are necessarily very drastic. I would spend more time looking at the format now, but I feel that is better saved for when we are actually formatting this data to be more usable on the whole.

Also, this commit makes _several breaking changes_ to the coop mode (such as completely removing the ability for the blocks to be draggable). See #47.

Sample output (from the `ibmdb:timeandlocation` column) with newlines added for readability:

```
{
     "Gesture 03/20/2018 18:52:27 (15.8664%,50.3074%)",
     "Instruction 03/20/2018 18:52:29 03/20/2018 18:52:30 1222 D",
     "Movement Block id: block2 Letter: D Color: blue 03/20/2018 18:52:30 (446.717px,88.0333px) (8.380554577136298,50.43667115764625)",
     "Instruction 03/20/2018 18:52:32 03/20/2018 18:52:34 1479 Flip D",
     "Flip Block id: block2 Letter: D Color: blue 03/20/2018 18:52:34 (8.380554577136298,50.43667115764625)"
}
```

### CODE REVIEW

File: 2D Simulation/index.js [hunk 1]
```diff
@@ -177,20 +168,21 @@ function initLetters(possibleLetters, numBlocks) {
     return lets;
 }
 
-function flipBlock(box, event, config) {
+function flipBlock(block_id, letter, color, config) {
```
Comment (jcolemang) [inline 2D Simulation/index.js:171]: Why are letter and color parameters here? I think that the block's letter and color should be a part of the configuration passed in

File: 2D Simulation/movesTracker.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    this.export_actions = function () {
+        let action_strs = [];
+
+        for (let action of this.actions) {
+            action_strs.push(action.export_func(action));
+        }
+
+        return action_strs;
+    }
+}
```
Comment (jcolemang) [inline 2D Simulation/movesTracker.js:99]: Usually when I've seen this module pattern used the function has been immediately applied. When are the fields in `movesTracker` actually being set?

Comment (jcolemang) [reply 2D Simulation/movesTracker.js:99]: The internet cleared it up for me.

**Review by jcolemang — APPROVED** [review]
I didn't read through the removed code very closely, but most of the added code looks fine to me.

**Comment by kgroble** [comment]
Seems good, definitely better organized now and net removal is always nice.

**Review by kgroble — APPROVED** [review]

**Pull request merged**